protocol/binary/stream_reader.go

```diff
@@ -66,24 +66,51 @@ type StreamReader struct {
 
 	reader io.Reader
 	buffer [8]byte
+
+	// discard is points to either discardOffset or discardStream based on
```

Comment:
nit: "discard ~~is~~ points"
```suggestion
	// discard points to either discardOffset or discardStream based on
```